### PR TITLE
.github: Delegate repo permissions to teams, instead of individuals

### PR DESCRIPTION
**Pull request by justaugustus**
Fixes https://github.com/todogroup/outbound-oss/issues/53.

Signed-off-by: Stephen Augustus <foo@auggie.dev>

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -4,9 +4,8 @@
 # the repo. Unless a later match takes precedence,
 # the following users/teams will be requested for
 # review when someone opens a pull request.
-# TODO(.github): Individual maintainers should be managed as a team
-* @todogroup/steering-committee @cornelius @misappi @mkurzman @OliverFendt
+* @todogroup/outbound-oss-maintainers
```
Comment (justaugustus) [inline .github/CODEOWNERS:7]: @cornelius @misappi @mkurzman @OliverFendt have all been added to this group.

File: .github/settings.yml [hunk 2]
```diff
@@ -12,26 +12,14 @@ repository:
...
-  - username: cornelius
-    permission: maintain
-
-  - username: mkurzman
-    permission: maintain
-
-  - username: OliverFendt
+  - name: outbound-oss-maintainers
     permission: maintain
+  - name: outbound-oss
```
Comment (justaugustus) [inline .github/settings.yml:24]: (Parent team w/ `triage` access)

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -4,9 +4,8 @@
...
 
 # Enforces admin protections for repo configuration via probot settings app.
 # ref: https://github.com/probot/settings#security-implications
-.github/settings.yml @todogroup/steering-committee
+.github/settings.yml @todogroup/outbound-oss-admins
```
Comment (justaugustus) [inline .github/CODEOWNERS:11]: @anajsana and I are part of this group.

**Review by justaugustus — COMMENTED** [review]
Added some comments to explain who is part of which team.

**Comment by geekygirldawn** [comment]
This looks good to me, thank you!

**Review by misappi — APPROVED** [review]
Looks good to me. Thanks.

**Comment by justaugustus** [comment]
Thanks folks. Merging this!

**Pull request merged**